File: Problems/Problem_2/problem_2.c
#include <stdio.h>
#include <math.h>

int fibonacci(int n){
    if (n==1 || n==2){
        return n-1;
    }
    else{
        return pow(fibonacci(n-1),n-1) + pow(fibonacci(n-2),n-2);
    }
}

int main(void){
    int N;
    printf("\nRecursive fibonacci relation = f(n) = f(n-1)^(n-1) + f(n-2)^(n-2)\n\n");
    printf("Please enter a number (Nth term of the sequence): ");
    scanf("%d", &N);
    int result;
    result = fibonacci(N);
    printf("f(%d) = %d\n", N, result);

    return 0;
}

File: Problems/Problem_2/problem2.java
import java.util.Scanner;


class Main{
	static double rec(double n){
		if(n == 1 || n == 2){
			return n-1;
		}
		else{
			return (Math.pow(rec(n-1), (n-1)) + Math.pow(rec(n-2), (n-2)));
		}
	}


    public static void main(String args[]){
    	Scanner myObj = new Scanner(System.in);
    	System.out.println("Enter n");
	
    	double n = myObj.nextDouble();
    	System.out.println(n);
    	System.out.println(rec(n));
	
    }
}